platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/schemaBuilder/schemaBuildingUtils.kt

```diff
@@ -36,17 +39,14 @@ fun KClassifier.toDataTypeRef(): DataTypeRef =
         Boolean::class -> DataTypeInternal.DefaultBooleanDataType.ref
         Long::class -> DataTypeInternal.DefaultLongDataType.ref
         is KClass<*> -> DataTypeRefInternal.DefaultName(DefaultFqName.parse(checkNotNull(qualifiedName)))
-        else -> error("unexpected type")
+        else -> error("can't convert unexpected type to data type reference: ${this::class.simpleName}")
```

Comment:
`this::class` is going to be something like `KTypeParameterImpl::class`, is it intended that this message uses its name and not the string representation of `this`? If it is, some clarification might be useful that it's the kind of the type used, not the type itself, as otherwise it might be confusing if reported like:

> can't convert unexpected type to data type reference: KTypeParameterImpl

, while in fact you don't use KTypeParameterImpl anywhere in your types.

Reply:
I'm not sure how to do this better then.

Reply:
Ok, pushed, please check it out.